packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx

```diff
@@ -53,15 +49,35 @@ const AccountRecoveryConfirmKey = (_: RouteComponentProps) => {
   // so we set its value after the first request for subsequent requests.
   const [fetchedResetToken, setFetchedResetToken] = useState('');
   const [isFocused, setIsFocused] = useState(false);
+  // For initial page load
+  const [showLoadingSpinner, setShowLoadingSpinner] = useState(true);
   // We use this to debounce the submit button
   const [isLoading, setIsLoading] = useState(false);
   const account = useAccount();
   const ftlMsgResolver = useFtlMsgResolver();
-  const { linkStatus, setLinkStatus, requiredParams } =
-    useAccountRecoveryConfirmKeyLinkStatus();
   const location = useLocation();
   const navigate = useNavigate();
 
+  useEffect(() => {
+    const checkPasswordForgotToken = async (token: string) => {
+      try {
+        const isValid = await account.resetPasswordStatus(token);
+        if (isValid) {
+          setLinkStatus(LinkStatus.valid);
+
+          setShowLoadingSpinner(false);
```

Comment:
Continuation from [this thread](https://github.com/mozilla/fxa/pull/15397#discussion_r1213773727):

Interesting - I would think that since React batches state updates this wouldn't happen, but it makes sense that the expired/damaged link state is handled in the HOC so the only time we really _have_ to set this to `false` is when we want to render the page. Maybe you could modify your `// For initial page load` comment to something like `// Show loading spinner until token is valid, else LinkValidator handles invalid states` so it's clear when we look at this later?